### PR TITLE
Enhance/migrate vue3

### LINKED ISSUE #1051: [Feature Request]: add filtering for esp-idf examples wizrd  (VSC-1197)
the examples wizard in the esp-idf is massive but doesn't contain any ability (that i know of) to actually look for a specfic example or filter for a subset of examples .

it would make a **huge** improvment to have at least a text box that would filter upon typing all the examples that contain a specific word , the same way the settings window contains an option to filter for specific settings. 
additional improvments can be 
* having the tree itself collapse or exampnded (again like the tree in the settings dialog) 
* have a combo box that displays only examples for specific targets (I.E s3,s2 etc..) 

only alternatives is actualy go to the github repo

### PR DESCRIPTION
**Pull request by brianignacio5**
## Description

Update web views from Vue 2 and vuex to Vue 3 and Pinia because vue 2 support will be dropped the end of this year.

Fix #1051 Added search for `ESP-IDF: Show Examples` command

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Steps to test this pull request

This PR is very big and many features need to be tested. Basically all web views under `src/views`

1. Right click a `CMakeLists.txt` file and select `ESP-IDF: CMakeLists.txt editor` command. If the file is root directory choose Project `Project CMakeLists.txt` otherwise choose `Component CMakeLists.txt`. Use UI to modify values and see changes on CMakeLists.txt. Run `Developer: Open web view Developer tools` and see if there is any error in the console.
2. Execute `ESP-IDF: Show examples projects`. Choose example, check rendered markdown view,  create project. Run `Developer: Open web view Developer tools` and see if there is any error in the console.
3. Execute `ESP-IDF: SDK Configuration editor (menuconfig)` command. Modify different kind of values like number, string, hex, checkbox, select dropdown. Check the ESP-IDF output for any error. Run `Developer: Open web view Developer tools` and see if there is any error in the console.
4. Execute `ESP-IDF: New Project` command. Modify all options, Try to add a local component, choose template and create a project. Run `Developer: Open web view Developer tools` and see if there is any error in the console.
5. Open an example with NVS data like Protocols -> http_server -> file_serving. Right click `nvs_data.csv` and select `ESP-IDF: Open NVS Partition editor`. add, Modify or remove values and types and click the different action buttons. Run `Developer: Open web view Developer tools` and see if there is any error in the console.
6. Open an example with Partition table data like Protocols -> http_server -> file_serving. Right click `partitions_example.csv` and select `ESP-IDF: Open Partition table editor UI`. add, Modify or remove values and types and click the different action buttons. Run `Developer: Open web view Developer tools` and see if there is any error in the console.
7. Open an example with multiple configuration like  Build system -> CMake -> multi_config and follow the [multiple config tutorial](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/tutorial/multiple_config.md) Run `Developer: Open web view Developer tools` and see if there is any error in the console.
8. Run the `ESP-IDF: Configure ESP-IDF extension` command and run the 3 kinds of setup from UI: Express Advanced and use existing setup. Run `Developer: Open web view Developer tools` and see if there is any error in the console.
9. Follow the [app tracing tutorial](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/tutorial/app_tracing.md) to test the Application tracing. Run `Developer: Open web view Developer tools` and see if there is any error in the console.
10. Follow the [heap tracing tutorial](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/tutorial/heap_tracing.md) to test the Heap Tracing. Run `Developer: Open web view Developer tools` and see if there is any error in the console.
11.  Run the `ESP-IDF: Welcome` to open the extension welcome page and test all links and buttons. Run `Developer: Open web view Developer tools` and see if there is any error in the console.

- Expected behaviour:

- Expected output:

## How has this been tested?

Manual testing of each view

**Test Configuration**:
* ESP-IDF Version: 5.0
* OS (Windows,Linux and macOS): macOS

## Dependent components impacted by this PR:

- Component 1
- Component 2

## Checklist
- [x] PR Self Reviewed
- [x] Applied Code formatting
- [x] Added Documentation
- [ ] Added Unit Test
- [ ] Verified on all platforms - Windows,Linux and macOS

### CODE REVIEW

File: src/views/nvs-partition-table/components/Row.vue [hunk 1]
```diff
@@ -150,3 +109,5 @@ export default class Row extends Vue {
   background-color: rgba(176, 81, 41, 0.1);
 }
 </style>
+
```
Comment (radurentea) [inline src/views/nvs-partition-table/components/Row.vue:112]: Extra empty lines

**Review by radurentea — CHANGES_REQUESTED** [review]
Left comments for small left overs. Also, the things found by code scanner should be addressed before merging

**Comment by AndriiFilippov** [comment]
@brianignacio5 hi ! 

```
1. Right click a CMakeLists.txt file and select ESP-IDF: CMakeLists.txt editor command. If the file is root directory choose Project Project CMakeLists.txt otherwise choose Component CMakeLists.txt. Use UI to modify values and see changes on CMakeLists.txt. Run Developer: Open web view Developer tools and see if there is any error in the console.
```

I think we should do some check here, caz it is possible to create multiple configurations for every option:
Not sure if it is OK 

image

**Comment by AndriiFilippov** [comment]
@brianignacio5 

it's tiring to search using scroll, we should definitely have filter for project examples: 

image

**Comment by AndriiFilippov** [comment]
@brianignacio5 hi!

1) I think we should add default (namespace) nvs row, like this: 
image

because it is fully empty now.
we have it in IDE:
image


2) For Type : namespace - lets delete drop-down window because it is empty field(confusing):
image

**Comment by kolipakakondal** [comment]
> @brianignacio5 hi! Tested: OS - Windows 10
> 
> Shouldn't this text be active link ? caz when I put cursor on this text - it reacts, but I can't click it
> 
> image

Please show all hyper links text in the blue color
Screenshot 2023-11-21 at 2 19 06 PM

**Comment by AndriiFilippov** [comment]
Brian, please, explain to me if the "Welcome Page" will be displayed once user install extension from "Extensions" ? 

Because now once I install extension using VSIX -> there is no "Welcome Page" displayed. I should do it manually.

**Comment by AndriiFilippov** [comment]
@brianignacio5 , hi! 

Should both commands have same structure ? 

image

**Comment by AndriiFilippov** [comment]
@brianignacio5 hi! 

when click "Save" button - there is no info message "Saved successfully" or any other info. I don't understand if it actually saved. 
Can be confusing for user.

image

**Comment by AndriiFilippov** [comment]
@brianignacio5, is it possible to add real time validation, caz now it works this way -> validation starts after pressing "Save" button.
So, for example in this case, I have already updated the field with valid value, but it still showing error:
Could be confusing for user.


image

**Comment by AndriiFilippov** [comment]
@brianignacio5 hi ! 

Getting some weird notification during 'Extension: Install from VSIX...'
Do you know what could be the reason ? 


image

**Comment by AndriiFilippov** [comment]
@brianignacio5 

Could we move search a bit left to make it more organic ?
image

**Comment by AndriiFilippov** [comment]
> @brianignacio5 hi !
> 
> ```
> 1.Right click a CMakeLists.txt file and select ESP-IDF: CMakeLists.txt editor command. If the file is root directory choose Project Project CMakeLists.txt otherwise choose Component CMakeLists.txt. Use UI to modify values and see changes on CMakeLists.txt. Run Developer: Open web view Developer tools and see if there is any error in the console.
> ```
> 
> we should do some check here, caz it is possible to create multiple configurations for every option:
> 
> image

I think we should maybe control this sequence somehow:
image

Because I am able to do CMakeList in any sequence:

image

**Comment by AndriiFilippov** [comment]
> @brianignacio5 hi!
> 
> 1. I think we should add default (namespace) nvs row, like this:
>    image
> 
> because it is fully empty now. we have it in IDE: image
> 
> 2. For Type : namespace - lets delete drop-down window because it is empty field(confusing):
>    image

Still empty

image

**Comment by brianignacio5** [comment]
> @brianignacio5 hi !
> 
> Getting some weird notification during 'Extension: Install from VSIX...' Do you know what could be the reason ?
> 
> image

Can you provide the extension log ?

**Comment by AndriiFilippov** [comment]
> > @brianignacio5 hi !
> > ```
> > 1.Right click a CMakeLists.txt file and select ESP-IDF: CMakeLists.txt editor command. If the file is root directory choose Project Project CMakeLists.txt otherwise choose Component CMakeLists.txt. Use UI to modify values and see changes on CMakeLists.txt. Run Developer: Open web view Developer tools and see if there is any error in the console.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > we should do some check here, caz it is possible to create multiple configurations for every option:
> > image
> 
> I think we should maybe control this sequence somehow: image
> 
> Because I am able to do CMakeList in any sequence:
> 
> image

@brianignacio5 hi ! 

I forgot to add link. 
https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/build-system.html#mandatory-parts

The optional variables: 
image

I do see structure in CMake editor: 
image

but once I open CMakeList.txt I do see the disordered structure: 

image

**Comment by AndriiFilippov** [comment]
@brianignacio5 

open CMakeList.txt -> delete everything, only Project Name and Minimum Required 
image

add 'Set variable with value set (VARIABLE "VALUE")  -> fill and save -> it disappears from CMake Editor but could see it in CMakeList.txt ->

image
image

**Comment by AndriiFilippov** [comment]
@brianignacio5 

open CMakeList.txt -> delete everything, only Project Name and Minimum Required

image

add Include PATH first -> next add  'Set variable with value set (VARIABLE "VALUE") like this ->

image

save -> open CMakeList -> wrong sequence ->

image

**Comment by AndriiFilippov** [comment]
@brianignacio5 , please, add message during tool installation via Espressif server, as we discussed before.

**Comment by AndriiFilippov** [comment]
@brianignacio5 hi ! 

tick on "Show Welcome on extension startup" does not work. ESP Welcome windows appears on every VS Code start. 
This is good advertising for us, but I’m afraid it will annoy users 🤪 

image

**Comment by AndriiFilippov** [comment]
@brianignacio5 

Windows 10 
ESP-IDF: v5.1.1 

after project creation -> build -> error 

`
[0/1] Re-running CMake...
-- ccache will be used for faster recompilation
-- Building ESP-IDF components for target esp32
CMake Error at C:/Users/AndriiFilippov/e/esp-idf-v5.1.1/tools/cmake/build.cmake:540 (message):
Call Stack (most recent call first):
  C:/Users/AndriiFilippov/e/esp-idf-v5.1.1/tools/cmake/project.cmake:547 (idf_build_process)
  CMakeLists.txt:6 (project)


-- Configuring incomplete, errors occurred!
See also "C:/Users/AndriiFilippov/e/esp-idf-v5.1.1/examples/get-started/hello_world/build/CMakeFiles/CMakeOutput.log".
←[31mFAILED: ←[0mbuild.ninja
C:\Users\AndriiFilippov\.espressif\tools\cmake\3.24.0\bin\cmake.exe --regenerate-during-build -SC:\Users\AndriiFilippov\e\esp-idf-v5.1.1\examples\get-started\hello_world -BC:\Users\AndriiFilippov\e\esp-idf-v5.1.1\examples\get-started\hello_world\build
ninja: error: rebuilding 'build.ninja': subcommand failed

 *  The terminal process "C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -Command ninja " terminated with exit code: 1. 
`

**Comment by AndriiFilippov** [comment]
> @brianignacio5
> 
> open CMakeList.txt -> delete everything, only Project Name and Minimum Required image
> 
> add 'Set variable with value set (VARIABLE "VALUE") -> fill and save -> it disappears from CMake Editor but could see it in CMakeList.txt ->
> 
> image image

still persist. 
Open CMake editor -> add 'Set variable with value' -> fin and save -> disappears from editor.

**Comment by AndriiFilippov** [comment]
@brianignacio5 

able to save CMake editor with Mandatory Parts such as Project Name or Minimum Requirement deleted.  

image

**Comment by brianignacio5** [comment]
> > @brianignacio5
> > open CMakeList.txt -> delete everything, only Project Name and Minimum Required image
> > add 'Set variable with value set (VARIABLE "VALUE") -> fill and save -> it disappears from CMake Editor but could see it in CMakeList.txt ->
> > image image
> 
> still persist. Open CMake editor -> add 'Set variable with value' -> fin and save -> disappears from editor.

What is `fin and save` ?

**Comment by AndriiFilippov** [comment]
> > @brianignacio5
> > open CMakeList.txt -> delete everything, only Project Name and Minimum Required image
> > add 'Set variable with value set (VARIABLE "VALUE") -> fill and save -> it disappears from CMake Editor but could see it in CMakeList.txt ->
> > image image
> 
> still persist. Open CMake editor -> add 'Set variable with value' -> fill and save -> disappears from editor.

issue still persist.

**Comment by brianignacio5** [comment]
> > > @brianignacio5
> > > open CMakeList.txt -> delete everything, only Project Name and Minimum Required image
> > > add 'Set variable with value set (VARIABLE "VALUE") -> fill and save -> it disappears from CMake Editor but could see it in CMakeList.txt ->
> > > image image
> > 
> > 
> > still persist. Open CMake editor -> add 'Set variable with value' -> fill and save -> disappears from editor.
> 
> issue still persist.

Should be fixed with latest commit.

**Comment by AndriiFilippov** [comment]
> @brianignacio5 , hi!
> 
> Should both commands have same structure ?
> 
> image

Please, fix this one

**Comment by AndriiFilippov** [comment]
> @brianignacio5 hi!
> 
> when click "Save" button - there is no info message "Saved successfully" or any other info. I don't understand if it actually saved. Can be confusing for user.
> 
> image

Please, fix this one

**Comment by AndriiFilippov** [comment]
@brianignacio5 hi ! 

Could you please explain to me the meaning of "Retry" button ? 

image

**Comment by brianignacio5** [comment]
> > @brianignacio5 hi!
> > when click "Save" button - there is no info message "Saved successfully" or any other info. I don't understand if it actually saved. Can be confusing for user.
> > image
> 
> Please, fix this one

This will fix in another PR. Not related to changes in this PR (purely UI )



> @brianignacio5 hi !
> 
> Could you please explain to me the meaning of "Retry" button ?
> 
> image

This is to reload the partition table content from the CSV (kinda like a reset button)

**Comment by AndriiFilippov** [comment]
@brianignacio5 hi ! 

Tested under: 
OS - Windows 10 
ESP-IDF:  v5.1.2 

LGTM 👍

**Pull request merged**